scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/daemon/ClusterSummary.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.scheduling.scheduler.engine.daemon;
+
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot;
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot.ContainerInstance;
+import com.amazonaws.blox.scheduling.state.ClusterSnapshot.Task;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+class ClusterSummary {
+  private final ClusterSnapshot snapshot;
+  private final Map<String, List<Task>> tasksByInstanceArn;
+  private final Map<String, ContainerInstance> instancesByInstanceArn;
+
+  public ClusterSummary(ClusterSnapshot snapshot) {
```

Comment:
when is this created?

Reply:
This wrapper is created in the abstract `DaemonScheduler` class's `schedule` template method.

The idea is that ReplaceAfterTerminate and TerminateAfterReplace will both extend that abstract class with the specific workflow logic, but reuse these same wrappers to work with the flat ClusterSnapshot data.